### ISSUE TITLE
Cookie is not set if HTTP proxy is used

### ISSUE THREAD

**Issue by j16sdiz** [franela/goreq#97]
As title.
If you set a HTTP_PROXY environment variable, cookies cannot be set.
This works on stock client

**Comment by marcosnils** [comment]
@j16sdiz I couldn't reproduce the issue you're seeing

Here you have a sample program which I've used to simulate your problem:

```golang
package main

import (
	"fmt"
	"net/http"
	"sync"

	"github.com/franela/goreq"
)

func main() {
	var lastReq *http.Request

	var wg sync.WaitGroup

	go func() {
		http.ListenAndServe(":1111", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			if r.Method == "GET" && r.URL.Path == "/" {
				lastReq = r
				w.Header().Add("x-forwarded-for", "test")
				w.WriteHeader(200)
				w.Write([]byte(""))
				wg.Done()
			} else if r.Method == "GET" && r.URL.Path == "/redirect_test/301" {
				http.Redirect(w, r, "/", 301)
			}
		}))
	}()

	wg.Add(1)

	goreq.Request{Uri: "http://www.google.com"}.WithCookie(&http.Cookie{Name: "Foo", Value: "Bar"}).Do()
	wg.Wait()

	fmt.Println(lastReq.Cookies())

}

```

If you run this program like this: `HTTP_PROXY=http://127.0.0.1:1111 go run main.go` you'll see that cookies are printed at the end of the program. So it seems like `goreq` is actually passing cookies when using proxy.

**Comment by j16sdiz** [comment]
What I means is, when the server return a `Set-Cookie` header, the client can't capture that correctly.

Step to reproduce:
1. Setup a proxy server. 
2. Get a page that have secure cookie (e.g. https://accounts.google.com/ , it set a "GAPS" cookie with Secure flag)
3. `jar.GetCookie(google_url)` is empty if proxy server is used. It have a "GAPS" cookie if no proxy is used.

The `Set-Cookie` header can be seen on the Response object. It is just the CookieJar don't have that.

**Comment by marcosnils** [comment]
@j16sdiz I see. I'll fix it today. 

Thanks for the feedback

**Comment by marcosnils** [comment]
@j16sdiz I apologize for the delay, I've been quite busy lately. 

I've setup a new example where a server returns the Set-Cookie option and I could retrieve it by either using the `response.Cookies()` method or directly inspecting the `CookieJar` used in the request

```go
func main() {
        var lastReq *http.Request

        go func() {
                http.ListenAndServe(":1111", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
                        if r.Method == "GET" && r.URL.Path == "/" {
                                lastReq = r 
                                w.Header().Set("Set-Cookie", "foo=bar")
                        }
                }))
        }()

        jar, _ := cookiejar.New(nil)
        purl, _ := url.Parse("http://localhost:1111/")                                                                                                                                                                                       
        res, err := goreq.Request{Uri: purl.String(), CookieJar: jar}.Do()
        fmt.Println(res.StatusCode, err)

        fmt.Println(res.Cookies())

        fmt.Println(jar.Cookies(purl))
}
```
Can you please provide a concrete example of the functionality you're seeing it's not present?

Thx!.

**Comment by marcosnils** [comment]
@j16sdiz I could reproduce the issue as you stated.  I've already applied a fix here 41f8ae0dfac5ef5d94708e02dfc19fab3e66bbf4

Please upgrade your goreq dependency and let us knows if it works. Thanks for pointing the issue!.
